packages/shared/apis/host/UserActionModule.ts

```diff
@@ -28,36 +27,39 @@ export function registerUserActionModuleServiceServerImplementation(port: RpcSer
         return {}
       }
 
-      let sceneThumbnailUrl: string | undefined
-      let sceneName: string = destination
-      let sceneCreator: string = 'Unknown'
       let sceneEvent = {}
+      const sceneData = {
+        name: 'Unnamed',
+        owner: 'Unknown',
+        previewImageUrl: ''
+      }
 
       const mapSceneData = (await fetchScenesByLocation([destination]))[0]
 
-      const metadata: Scene | undefined = mapSceneData?.entity.metadata
+      if (mapSceneData) {
+        const metadata: Scene | undefined = mapSceneData?.entity.metadata
 
-      sceneName = getSceneName(destination, metadata)
-      sceneCreator = getOwnerNameFromJsonData(metadata)
+        sceneData.name = postProcessSceneName(getSceneNameFromJsonData(metadata))
+        sceneData.owner = getOwnerNameFromJsonData(metadata)
+        sceneData.previewImageUrl =
+          getThumbnailUrlFromJsonDataAndContent(
+            mapSceneData.entity.metadata,
+            mapSceneData.entity.content,
+            getFetchContentServer(store.getState())
+          ) || sceneData.previewImageUrl
 
-      if (mapSceneData) {
-        sceneThumbnailUrl = getThumbnailUrlFromJsonDataAndContent(
-          mapSceneData.entity.metadata,
-          mapSceneData.entity.content,
-          getFetchContentServer(store.getState())
-        )
+        if (!sceneData.previewImageUrl) {
+          let sceneParcels = [destination]
+          if (metadata && metadata.scene.parcels) {
+            sceneParcels = metadata.scene.parcels
+          }
+          sceneData.previewImageUrl = `https://api.decentraland.org/v1/map.png?width=480&height=237&size=10&center=${destination}&selected=${sceneParcels.join(
```

Comment:
Are we using this also for the dev environment? If yes, can we move this to an env var per environment?

Reply:
solved